src/bitcoin-cli.cpp

```diff
@@ -34,7 +35,7 @@ std::string HelpMessageCli()
     strUsage += HelpMessageOpt("-datadir=<dir>", _("Specify data directory"));
     AppendParamsHelpMessages(strUsage);
     strUsage += HelpMessageOpt("-rpcconnect=<ip>", strprintf(_("Send commands to node running on <ip> (default: %s)"), DEFAULT_RPCCONNECT));
-    strUsage += HelpMessageOpt("-rpcport=<port>", strprintf(_("Connect to JSON-RPC on <port> (default: %u or testnet: %u)"), BaseParams(CBaseChainParams::MAIN).RPCPort(), BaseParams(CBaseChainParams::TESTNET).RPCPort()));
+    strUsage += HelpMessageOpt("-rpcport=<port>", strprintf(_("Connect to JSON-RPC on <port> (default: %u)"), defaultBaseParams->RPCPort()));
```

Comment:
Regarding looping and not wanting to show regtest's information, we could easily do that just by leaving regtest out of supportedChains (see https://github.com/jtimon/bitcoin/commit/160d874dfdb53c71a07b1bf7bdd0dab571dabefa#diff-64cbe1ad5465e13bc59ee8bb6f3de2e7R20 ). Although that would also leave it out from https://github.com/jtimon/bitcoin/commit/160d874dfdb53c71a07b1bf7bdd0dab571dabefa#diff-de33fb0129cd5121c7098035527b26fdR14 and https://github.com/jtimon/bitcoin/commit/28d961ff939bfc0887c6c8b8b288733864559abd , which is probably not a big deal. The special case for testnet3 can be solved by replacing a "main and testnet" enumaration whether we loop around 2 or 3 elements.
Just a thought. Or we can just leave it as it is, it's just I thought that I had when I was paranoid about https://github.com/jtimon/bitcoin/commit/28d961ff939bfc0887c6c8b8b288733864559abd maybe being controversial (as a "too altcoinish" commit) and I just remembered.